pkg/io/opener.go

```diff
@@ -0,0 +1,122 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package io
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"os"
+	"strings"
+
+	"cloud.google.com/go/storage"
+	"github.com/sirupsen/logrus"
+	"google.golang.org/api/option"
+
+	"k8s.io/test-infra/testgrid/util/gcs" // TODO(fejta): move this logic here
+)
+
+type storageClient interface {
+	Bucket(name string) *storage.BucketHandle
+}
+
+// Aliases to types in the standard library
+type (
+	ReadCloser  = io.ReadCloser
+	WriteCloser = io.WriteCloser
```

Comment:
Ah I see now that it is because this package is also called `io`.  It seems a bit strange to use the same package name as a standard library package, but I guess that makes sense here.